pkg/health/server.go

```diff
@@ -33,16 +34,56 @@ type Server interface {
 }
 
 type server struct {
-	ready *atomic.Bool
-	log   logger.Logger
+	ready  *atomic.Bool
+	router http.Handler
+	log    logger.Logger
+}
+
+type RouterOptions func(log logger.Logger) (string, http.Handler)
+
+func NewRouterOptions(path string, handler http.Handler) RouterOptions {
+	return func(log logger.Logger) (string, http.Handler) {
+		return path, handler
+	}
+}
+
+func NewJSONDataRouterOptions[T any](path string, getter func() (T, error)) RouterOptions {
```

Comment:
I don't know if a generic function is really needed here, since we are not treating the data any differently depending on the type. It may be more efficient to not use generics:


```suggestion
func NewJSONDataRouterOptions(path string, getter func() (any, error)) RouterOptions {
```

Reply:
IMHO, generic function is needed here. `func() Any-specific-type` have a different signature than `func() any`. For example, this couldn't be compiled `var f func() any = func() int { return 1 }`.

For performance, we are not using interface constrain here, I think the go compiler would be clever enough monomorphizing the code -- generates a copy of the generic function for the specific data type it is invoked with.

Reply:
Ok I see what you mean now.